CMake/Dependencies.cmake

```diff
@@ -3,6 +3,21 @@
 include(functions/dependency_options)
 include(functions/emscripten_system_library)
 
+if(EMSCRIPTEN)
+  emscripten_system_library("zlib" ZLIB::ZLIB USE_ZLIB=1)

```

Comment:
I don't think we need `zlib` on emscripten. I'm guessing you need it here because of the precompiled SDL2?

Reply:
it was already present in the CMake file and i just moved to the beginning. and i need the zlib as there are no package managers for the build as vcpkg was causing incompatibilities with its packages